tst/TerminalCategory.tst

```diff
@@ -0,0 +1,1918 @@
+##################################################################################
+# Test the correspondences between the cartesian closed structure of the category
+# of the terminal category and the cocartesian coclosed structure of its opposite.
+##################################################################################
+
+gap> START_TEST("TerminalCategory.tst");
+
+##############################################################
+# (Co)Product, Exp and Coexp on objects
+##############################################################
+
+# Products and exponential objects for the terminal category
+
+gap> T := TerminalCategoryWithMultipleObjects( );;
+gap> a := ObjectConstructor( T, "a" );;
+gap> b := ObjectConstructor( T, "b" );;
+gap> c := ObjectConstructor( T, "c" );;
+gap> d := ObjectConstructor( T, "d" );;
+
+gap> i := InitialObject( T );;
+gap> t := TerminalObject( T );;
+
+gap> a_product_b := DirectProduct( a, b );;
+gap> b_product_a := DirectProduct( b, a );;
+gap> c_product_d := DirectProduct( c, d );;
+gap> d_product_c := DirectProduct( d, c );;
+
+gap> i_product_a := DirectProduct( i, a );;
+gap> a_product_i := DirectProduct( a, i );;
+
+gap> t_product_a := DirectProduct( t, a );;
+gap> a_product_t := DirectProduct( a, t );;
+
+gap> exp_ab := ExponentialOnObjects( a, b );;
+gap> exp_ba := ExponentialOnObjects( b, a );;
+gap> exp_cd := ExponentialOnObjects( c, d );;
+gap> exp_dc := ExponentialOnObjects( d, c );;
+
+gap> exp_ia := ExponentialOnObjects( i, a );;
+gap> exp_ai := ExponentialOnObjects( a, i );;
+gap> exp_ta := ExponentialOnObjects( t, a );;
+gap> exp_at := ExponentialOnObjects( a, t );;
+
+# Coproducts and coexponentials for opposite
+
+gap> T_op := Opposite( T );;
+
+gap> aop := Opposite( a );;
+gap> bop := Opposite( b );;
+gap> cop := Opposite( c );;
+gap> dop := Opposite( d );;
+gap> iop := Opposite( i );;
```

Comment:
I just noticed that this is redundant because of line 54 below. I'll fix it in FinSets.

Reply:
I agree.